File: 2014/FASE 1/C++/q-corretas-c++/carteiro.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N, M;
    cin >> N >> M;

    vector<int> houses(N);
    vector<int> orders(M);

    // Leitura das casas
    for (int i = 0; i < N; ++i) {
        cin >> houses[i];
    }

    // Leitura das encomendas
    for (int i = 0; i < M; ++i) {
        cin >> orders[i];
    }

    int time = 0;
    int current_house = 0;

    // Para cada encomenda
    for (int i = 0; i < M; ++i) {
        // Encontra a posição da casa onde a encomenda deve ser entregue
        int target_house = find(houses.begin(), houses.end(), orders[i]) - houses.begin();

        // Calcula o tempo necessário para ir até essa casa
        time += abs(target_house - current_house);

        // Atualiza a posição atual
        current_house = target_house;
    }

    cout << time << endl;

    return 0;
}

File: 2014/FASE 1/JAVA/q-corretas-java/carteiro.java
import java.util.Scanner;

public class carteiro {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        // Lê o número de casas e o número de encomendas
        int N = in.nextInt();
        int M = in.nextInt();

        // Lê os números das casas
        int[] casas = new int[N];
        for (int i = 0; i < N; i++) {
            casas[i] = in.nextInt();
        }

        // Lê os números das casas onde as encomendas devem ser entregues
        int[] encomendas = new int[M];
        for (int i = 0; i < M; i++) {
            encomendas[i] = in.nextInt();
        }

        // Calcula o tempo necessário para entregar todas as encomendas
        int tempo = 0;
        int posicaoAtual = 0; // Começa na primeira casa
        for (int i = 0; i < M; i++) {
            // Encontra a posição da casa onde a encomenda deve ser entregue
            int posicaoEncomenda = encontrarPosicao(casas, encomendas[i]);

            // Calcula o tempo necessário para chegar na casa da encomenda
            tempo += Math.abs(posicaoEncomenda - posicaoAtual);

            // Atualiza a posição atual
            posicaoAtual = posicaoEncomenda;
        }

        // Escreve o resultado
        System.out.println(tempo);
    }

    // Função para encontrar a posição de uma casa no array de casas
    public static int encontrarPosicao(int[] casas, int casa) {
        for (int i = 0; i < casas.length; i++) {
            if (casas[i] == casa) {
                return i;
            }
        }
        return -1; // Caso não encontre a casa (não deveria ocorrer)
    }
}

